server/lib/src/idm/server.rs

```diff
@@ -1340,13 +1338,13 @@ impl<'a> IdmServerAuthTransaction<'a> {
 
         if !slock.is_valid() {
             security_info!("Account is softlocked.");
-            //TODO: Did we want to return an error condition?
             return Ok(None);
         }
 
         let Ok(password) = cred.password_ref() else {
             // The credential should only ever be a password
-            unreachable!();
+            error!("user unix or primary credential is not a password");
```

Comment:
```suggestion
            error!("User's UNIX or primary credential is not a password, can't authenticate!");
```

Side note, perhaps for @Firstyear: I wonder if it's faster to check this first or the softlock with its read/write/change steps, given they can't auth at all...

Reply:
Faster yes, but not correct. In this case cred.password_ref() should in 99.99999% of cases always return a password. We just need to handle that fractional and unlikely case it doesn't since that's still a possibly until I undo some of the credential structure tech-debt. 

So I think in this case, we leave the logic as is because it makes more sense from the view of "okay, get the credential, are the softlocked, now lets check the password" and the unlikely case is ... well, unlikely.